### PR TITLE
crash when multiple effect blocks are concatenated

### PR DESCRIPTION
**Pull request by swilly22**

### CODE REVIEW

File: src/effects/effects.c [hunk 1]
```diff
@@ -247,6 +247,7 @@ EffectsBuffer *EffectsBuffer_New
 	void
 ) {
 	size_t n = 62500;  // initial size of buffer
+	//size_t n = 100000000;  // initial size of buffer
```
Comment (AviAvni) [inline src/effects/effects.c:250]: delete this?

File: src/effects/effects_apply.c [hunk 2]
```diff
@@ -86,9 +86,7 @@ static void ApplyCreateNode
 	//--------------------------------------------------------------------------
 
 	LabelID labels[lbl_count];
-	for(ushort i = 0; i < lbl_count; i++) {
-		fread_assert(labels + i, sizeof(LabelID), stream);
-	}
+	fread_assert(labels, lbl_count * sizeof(LabelID), stream);
```
Comment (AviAvni) [inline src/effects/effects_apply.c:89]: I think sizeof(labels) will do the multiplication

**Review by AviAvni — APPROVED** [review]
see 2 comments

File: src/graph/graph.c [hunk 3]
```diff
@@ -280,11 +280,13 @@ MATRIX_POLICY Graph_GetMatrixPolicy
 }
 
 // define the current behavior for matrix creations and retrievals on this graph
-void Graph_SetMatrixPolicy
+MATRIX_POLICY Graph_SetMatrixPolicy
 (
 	Graph *g,
 	MATRIX_POLICY policy
 ) {
+	MATRIX_POLICY prev_policy = Graph_GetMatrixPolicy(g);
```
Comment (AviAvni) [inline src/graph/graph.c:288]: in the first call probably this is not initialize

Comment (swilly22) [reply src/graph/graph.c:288]: When creating a new Graph via `Graph_New` the policy is set.

File: src/effects/effects_apply.c (hunk 2 above)
Comment (swilly22) [reply src/effects/effects_apply.c:89]: you're probably right.

**Pull request merged**